packages/ai-chat/src/common/chat-model.ts

```diff
@@ -570,6 +570,13 @@ export class ChangeSetImpl implements ChangeSet {
         return true;
     }
 
+    addOrReplaceElement(element: ChangeSetElement): void {
+        if (!this.replaceElement(element)) {
+            this.addElement(element);
+            this.notifyChange();
```

Comment:
I think we shouldn't notify again because addElement already notifies.

Reply:
True, I was totally confused about your previous comment that your change introduces a duplicate notification :-)